caproto/threading/client.py

```diff
@@ -1849,14 +1845,14 @@ def removed(_):
             # TODO: strong reference to non-instance methods?
             ref = weakref.ref(func, removed)
 
-        with self._callback_lock:
+        with self.callback_lock:
             cb_id = self._callback_id
             self._callback_id += 1
             self.callbacks[cb_id] = ref
         return cb_id
 
     def remove_callback(self, token):
-        with self._callback_lock:
+        with self.callback_lock:
             self.callbacks.pop(token, None)
```

Comment:
`pop` is atomic

Reply:
But you do not want this to fire while something else is iterating over the callbacks etc.

Reply:
Right.